tools/tests/com.liferay.ide.xml.search.ui.tests/src/com/liferay/ide/xml/search/ui/tests/LiferayHookXmlTests.java

```diff
@@ -368,9 +372,34 @@ protected void testServiceTypeAndServiceImplHyperlink()
     {
     }
 
-    // TODO
-    protected void testServiceTypeAndServiceImplValidation()
+    @Test
+    public void testServiceTypeAndServiceImplValidation() throws Exception
     {
+        // IDE-1810
+        if( shouldSkipBundleTests() )
+            return;
+
+        final IFile descriptorFile = getDescriptorFile();
+        final String elementName1 = "service-type";
+        final String elementName2 = "service-impl";
+
+        String elementValue1 = "com.liferay.portal.service.AccountLocalService";
+        String elementValue2 = "ExtAccountLocalServiceWrong";
+
+        setElementContent( descriptorFile, elementName1, elementValue1 );
+        setElementContent( descriptorFile, elementName2, elementValue2 );
+
+        String markerMessage =
+            MessageFormat.format( LiferayHookDescriptorValidator.MESSAGE_TYPE_NOT_FOUND, new Object[] { elementValue2 } );
+
+        buildAndValidate( descriptorFile );
+        assertEquals( true, checkMarkerByMessage( descriptorFile, MARKER_TYPE, markerMessage, true ) );
+
+        elementValue2 = "com.liferay.ide.tests.ExtAccountLocalService";
+        setElementContent( descriptorFile, elementName2, elementValue2 );
+
+        buildAndValidate( descriptorFile );
+        assertTrue( checkNoMarker( descriptorFile, MARKER_TYPE ) );
```

Comment:
Also in this same test we should set the value to something that is valid and then revalidate and make sure that there is now no more message related to this warning/error.